eo-runtime/src/test/java/EOorg/EOeolang/EOcageTest.java

```diff
@@ -277,6 +272,147 @@ private static void writeTo(final Phi cage, final Phi obj) {
         ).take(Boolean.class);
     }
 
+    /**
+     * Cases to test behaviour of cage with recursion.
+     * @since 0.1
+     */
+    @Nested
+    class RecursionTests {
+
+        /**
+         * DEPTH.
+         */
+        private static final int MAX_DEPTH = 50;
+
+        @BeforeEach
+        void setDepth() {
+            System.setProperty(
+                PhTracedEnclosure.MAX_CAGE_RECURSION_PROPERTY_NAME, String.valueOf(MAX_DEPTH)
+            );
+        }
+
+        @AfterEach
+        void clearDepth() {
+            System.clearProperty(PhTracedEnclosure.MAX_CAGE_RECURSION_PROPERTY_NAME);
+        }
+
+        @Test
+        void throwsExceptionIfRecursion() {
+            final Phi cage = new EOcage(Phi.Φ);
+            writeTo(cage, cage);
+            Assertions.assertThrows(
+                ExAbstract.class,
+                new Dataized(cage)::take,
+                "We expect the exception to be thrown since we have recursion here"
+            );
+        }
+
+        @Test
+        void doesNotThrowExceptionIfSmallDepth() {
+            final EOcage cage = new EOcage(Phi.Φ);
+            writeTo(
```

Comment:
@levBagryansky it would be better to use this method with with class prefix: `EOcageTest.writeTo(...)`

Reply:
done